pyproject.toml

```diff
@@ -289,7 +289,7 @@ generate-setup-file = false
 script = "build-module.py"
 
 [tool.poetry.extras]
-pyarrow = ["pyarrow"]
+pyarrow = ["pyarrow", "pyiceberg-core"]
```

Comment:
__technically__ this is not required, because we only use the transforms when writing to a partitioned table. But I think that it might lead to a lot of confusion if we don't do this.

Reply:
Agreed i was considering this too and came to the same conclusion :)

Also I think the `pyarrow_transform` functions are used on the read path too
https://github.com/apache/iceberg-python/blob/a67c5592f3243d255519581fedfcc5d93274b9c8/pyiceberg/io/pyarrow.py#L2696

Reply:
> Also I think the pyarrow_transform functions are used on the read path too

For completeness, I don't think that's true. `_determine_partitions` is only used by `_dataframe_to_data_files`. When reading, we often have a single value `SELECT * FROM tbl WHERE created_at > '2025-01-01 19:25:00'`, so there performance is not that important, and we use non-Arrow transform ([example for the MonthPartitioning](https://github.com/apache/iceberg-python/blob/f507dbd1c29c7d21d24c6f2f7b4479cde2be16de/pyiceberg/transforms.py#L529-L556)).